### PR TITLE
Implement `put_out_shares()`

### PR DESCRIPTION
**Pull request by nakatsuka-y**
This PR implements `put_out_shares()` which is used by Leader to merge processed share from itself and Helper after finishing the continuation phase.

A unit test that tests whether an initialization request has been sent before the continuation request has been sent.
`put_out_shares()` has also been integrated to the e2e test alongside the Leader sending the aggregation continuation request to Helper.

This partially implements #20 (continuation phase).

### CODE REVIEW

File: daphne/src/roles_test.rs [hunk 1]
```diff
@@ -469,4 +506,55 @@ async fn e2e() {
...
+        .vdaf
+        .produce_agg_init_req(
+            &leader,
+            &task_config.vdaf_verify_key,
+            task_id,
+            &agg_job_id,
+            reports,
+        )
+        .unwrap();
+    assert_matches!(transition, DapLeaderTransition::Continue(_, _));
```
Comment (cjpatton) [inline daphne/src/roles_test.rs:523]: A bit nicer:
```suggestion
    assert_matches!(transition, DapLeaderTransition::Continue(..));
```

**Review by cjpatton — CHANGES_REQUESTED** [review]
Tests:
* We should also add a test to make sure that sender auth is enforce for agg continue request. (Also for agg init request if missing.)
* For the e2e tests, it looks like you're manually constructiong the agg init and agg continue requests. Why not use `DapLeader::process()` to drive this instead?

File: daphne/src/roles_test.rs (hunk 1 above)
Comment (nakatsuka-y) [reply daphne/src/roles_test.rs]: Fixed! Thanks for the suggestion.

**Comment by nakatsuka-y** [comment]
Sender auth test for `AggregateContReq` was missing, so I've added it. This test for `AggregateInitReq` was already there.

`DapLeader::process()` has many function that we don't currently support, mainly the collection routines.
This is why I'm writing a separate e2e test.
At the end of the development cycle we should be able to run `process()`.

File: daphne/src/testing.rs [hunk 2]
```diff
@@ -212,7 +212,7 @@ impl DapAggregator<BearerToken> for MockAggregator {
             if let Some(agg_share) = agg_store.get_mut(&bucket_info) {
                 agg_share.merge(agg_share_delta)?;
             } else {
-                agg_store.insert(bucket_info.clone(), DapAggregateShare::default());
+                agg_store.insert(bucket_info.clone(), agg_share_delta);
```
Comment (cjpatton) [inline daphne/src/testing.rs:215]: Good catch :)

**Review by cjpatton — COMMENTED** [review]
> Sender auth test for `AggregateContReq` was missing, so I've added it. This test for `AggregateInitReq` was already there.
> 
> `DapLeader::process()` has many function that we don't currently support, mainly the collection routines. This is why I'm writing a separate e2e test. At the end of the development cycle we should be able to run `process()`.

ACK, sounds good! Go ahead and squash.

File: daphne/src/testing.rs (hunk 2 above)
Comment (nakatsuka-y) [reply daphne/src/testing.rs:215]: 😄

**Review by cjpatton — APPROVED** [review]

**Pull request merged**